### PR TITLE
Flow graph control flow blocks 2

### PR DESCRIPTION
**Pull request by carolhmj**
# Changes

- Add debounce and flip flop blocks
- About debounce: I'm not 100% sure the implementation should work this way, feedback is appreciated 🙂 

# Examples:
- #RKQX23#11: Debounce, only every 2 clicks on the sphere are registered
- #J1N2GP#7: Flip flop, click on the sphere to open and close the door

### CODE REVIEW

**Comment by bjsplat** [comment]
Please make sure to label your PR with "bug", "new feature" or "breaking change" label(s).
To prevent this PR from going to the changelog marked it with the "skip changelog" label.

**Comment by bjsplat** [comment]
Snapshot stored with reference name:
refs/pull/14355/merge

Test environment: 
https://babylonsnapshots.z22.web.core.windows.net/refs/pull/14355/merge/index.html

To test a playground add it to the URL, for example:

https://babylonsnapshots.z22.web.core.windows.net/refs/pull/14355/merge/index.html#WGZLGJ#4600

Links to test babylon tools with this snapshot:

https://playground.babylonjs.com/?snapshot=refs/pull/14355/merge
https://sandbox.babylonjs.com/?snapshot=refs/pull/14355/merge
https://gui.babylonjs.com/?snapshot=refs/pull/14355/merge
https://nme.babylonjs.com/?snapshot=refs/pull/14355/merge

To test the snapshot in the playground with a playground ID add it after the snapshot query string:

https://playground.babylonjs.com/?snapshot=refs/pull/14355/merge#BCU1XR#0

File: packages/dev/core/src/FlowGraph/Blocks/Execution/ControlFlow/flowGraphDebounceBlock.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+import { RichTypeNumber } from "core/FlowGraph/flowGraphRichTypes";
+import type { FlowGraphSignalConnection } from "core/FlowGraph/flowGraphSignalConnection";
+import { FlowGraphWithOnDoneExecutionBlock } from "core/FlowGraph/flowGraphWithOnDoneExecutionBlock";
+
+/**
+ * @experimental
+ * This block debounces the execution of a input, i.e. ensures that the input is only executed once every X times
+ */
+export class FlowGraphDebounceBlock extends FlowGraphWithOnDoneExecutionBlock {
+    public readonly count: FlowGraphDataConnection<number>;
```
Comment (sebavan) [inline packages/dev/core/src/FlowGraph/Blocks/Execution/ControlFlow/flowGraphDebounceBlock.ts:12]: Missing doc on this and the three following ones.

File: packages/dev/core/src/FlowGraph/Blocks/Execution/ControlFlow/flowGraphDebounceBlock.ts [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    constructor() {
+        super();
+        this.count = this._registerDataInput("count", RichTypeNumber);
+        this.reset = this._registerSignalInput("reset");
+        this.currentCount = this._registerDataOutput("currentCount", RichTypeNumber);
+    }
+
+    public _execute(context: FlowGraphContext, callingSignal: FlowGraphSignalConnection): void {
+        if (callingSignal === this.reset) {
+            context._setExecutionVariable(this, "debounceCount", 0);
```
Comment (sebavan) [inline packages/dev/core/src/FlowGraph/Blocks/Execution/ControlFlow/flowGraphDebounceBlock.ts:25]: I wonder if those should be constant or inside of a construtct preventing spelling mistakes.

Comment (sebavan) [reply packages/dev/core/src/FlowGraph/Blocks/Execution/ControlFlow/flowGraphDebounceBlock.ts:25]: This would be valid for all the nodes

Comment (carolhmj) [reply packages/dev/core/src/FlowGraph/Blocks/Execution/ControlFlow/flowGraphDebounceBlock.ts:25]: @sebavan I can do a pass during testing to move all these to constants?

Comment (sebavan) [reply packages/dev/core/src/FlowGraph/Blocks/Execution/ControlFlow/flowGraphDebounceBlock.ts:34]: lets add a task for it to not forget

**Review by sebavan — APPROVED** [review]

**Pull request merged**